uiowa/uiowa
# [Registrar] File archive block

# How to test

```
ddev blt ds --site=registrar.uiowa.edu && ddev drush @registrar.local uli /media/add/file
```
## Test session
1. Add a file and choose "End of Session Class Counts"
  - General Catalog, Past Miscellaneous, Tuition and fees should show the `date range` field
  - Census Report, Commencement Program, End of Session Class Counts, Iowa College and University Enrollment Report, Official Census Date Class Counts, Profile of Students Enrolled should show the `session` field. 
2. Add "File archive" block to `node/426/layout` and select "End of Session Class Counts" in the filter config. 
3. Save page and confirm that you can see and download the file listed in a table with the title "Session - File Archive Type" pattern. 
4. Test sorting - do they display newest date first?
5. Test editing the files you created and confirm the fields show/hide as expected. 

## Test date range
1. Add a file and choose "General Catalog"
  - General Catalog, Past Miscellaneous, Tuition and fees should show the `date range` field
  - Census Report, Commencement Program, End of Session Class Counts, Iowa College and University Enrollment Report, Official Census Date Class Counts, Profile of Students Enrolled should show the `session` field
2. Add "File archive" block to `/node/431/layout` and select "General Catalog" in the filter config
3. Save page and confirm that you can see and download the file listed in a table with the title "Date Range - File Archive Type" pattern
4. Test sorting - do they display newest date first?
5. Test editing the files you created and confirm the fields show/hide as expected. 

 ### Code review and questions
- Is setting up `SessionOptionsTrait` here the best way to create the allowed_values function for the session field?
  - How many previous sessions should we pull in?
   - As this updates each year older sessions will not be visible in the allowed values, will there be data loss?
- Do values need to be unset on the form if they are hidden?
